rpcserver.go

```diff
@@ -1722,6 +1722,43 @@ func (r *rpcServer) SignVirtualPsbt(_ context.Context,
 		return nil, fmt.Errorf("error decoding packet: %w", err)
 	}
 
+	// Make sure the input keys are known.
+	for _, input := range vPkt.Inputs {
+		// If we have all the derivation information, we don't need to
+		// do anything.
+		if len(input.Bip32Derivation) > 0 &&
+			len(input.TaprootBip32Derivation) > 0 {
+
+			continue
+		}
+
+		scriptKey := input.Asset().ScriptKey
+
+		// If the full tweaked script key isn't set on the asset, we
+		// need to look it up in the local database, to make sure we'll
+		// be able to sign for it.
+		if scriptKey.TweakedScriptKey == nil {
+			tweakedScriptKey, err := r.cfg.AssetWallet.FetchScriptKey(
+				ctx, scriptKey.PubKey,
```

Comment:
So we're making a distinction here between the internal key of the script key, and the final taproot output level public key? 

IIRC, this is for situations where some other related wallet knows the internal key, but _not_ the final tweaked version which might have an actual script leaf commitment?